integration/ingester_test.go

```diff
@@ -0,0 +1,336 @@
+						{
+							Timestamp: model.Time(queryStart.UnixMilli()),
+							Value:     model.SampleValue(100),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep).UnixMilli()),
+							Value:     model.SampleValue(110),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 2).UnixMilli()),
+							Value:     model.SampleValue(120),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 3).UnixMilli()),
+							Value:     model.SampleValue(130),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 4).UnixMilli()),
+							Value:     model.SampleValue(140),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 5).UnixMilli()),
+							Value:     model.SampleValue(150),
+						},
+					},
+				},
+			},
+		},
+		"native histogram series": {
+			inSeries: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{
+							Name:  "__name__",
+							Value: query,
+						},
+					},
+					Histograms: []prompb.Histogram{
+						remote.HistogramToHistogramProto(queryStart.UnixMilli(), test.GenerateTestHistogram(1)),
+						remote.HistogramToHistogramProto(queryStart.Add(queryStep).UnixMilli(), test.GenerateTestHistogram(2)),
+						remote.HistogramToHistogramProto(queryStart.Add(queryStep*2).UnixMilli(), test.GenerateTestHistogram(3)),
+						remote.HistogramToHistogramProto(queryStart.Add(queryStep*3).UnixMilli(), test.GenerateTestHistogram(4)),
+						remote.HistogramToHistogramProto(queryStart.Add(queryStep*4).UnixMilli(), test.GenerateTestHistogram(5)),
+						remote.HistogramToHistogramProto(queryStart.Add(queryStep*5).UnixMilli(), test.GenerateTestHistogram(6)),
+					},
+				},
+			},
+			expected: model.Matrix{
+				{
+					Metric: model.Metric(map[model.LabelName]model.LabelValue{"__name__": "foobar"}),
+					Histograms: []model.SampleHistogramPair{
+						{
+							Timestamp: model.Time(queryStart.UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(1),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep).UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(2),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 2).UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(3),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 3).UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(4),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 4).UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(5),
+						},
+						{
+							Timestamp: model.Time(queryStart.Add(queryStep * 5).UnixMilli()),
+							Histogram: test.GenerateTestSampleHistogram(6),
+						},
+					},
+				},
+			},
+		},
+		"series switching from float to native histograms at timestamp": {
+			inSeries: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{
+							Name:  "__name__",
+							Value: query,
```

Comment:
While the variable is the same, it is confusing to have `query` here (given the `expected` doesn't use `query` too)
```suggestion
							Value: "foobar",
```